src/libsyntax/parse/parser.rs

```diff
@@ -2936,6 +2936,18 @@ impl<'a> Parser<'a> {
                 self.err_dotdotdot_syntax(self.token.span);
             }
 
+            if self.token == token::LArrow {
+                self.diagnostic().struct_span_err(self.token.span, {
```

Comment:
```suggestion
                self.struct_span_err(self.token.span,
```

Reply:
I moved the error portion to a function called `err_larrow_operator` analogous to `err_dotdotdot_syntax`. I hope that's ok.